### PR TITLE
Refactor to use lang-extension-lib

### PR DESCRIPTION
**Pull request by JamesHovet**
**In short, this PR:**
- splits the python extension into two parts, the python-specific code (this repo) and common library code that handles creating a subprocess and passing messages to it (https://github.com/NetLogo/NetLogoExtensionLanguageServerLibrary). 
    - The above library was built by extracting existing code from this extension, so the underlying flow of data is essentially unchanged. 
- adds two new major features:
    - support for automatically serializing agents and agentsets in py:set calls. This makes passing NetLogo agent and agentset state to Python for analysis much simpler. 
    - a pop-out python REPL interpreter connected to the same Python environment as the main NetLogo window. With this pop-out panel you can quickly prototype python code in the NetLogo environment. 

**Background:**
The python extension has and continues to work by passing messages to a python sub process over localhost network calls. It was identified that the logic for creating these subprocess and passing serialized messages between them was independent of anything Python Specific and could just as easily apply to other languages. In an effort to make developing language interoperability extensions for NetLogo easier, the decision was made to separate this plumbing code into a separate library. 

**What is included in this PR:**
I undertook this task and extracted much of the code that was originally in this repo into a new library https://github.com/NetLogo/NetLogoExtensionLanguageServerLibrary. In so doing, I refactored it significantly and added the two features mentioned above. Since these features are part of the library code, they can be utilized by other language extensions at basically no cost. 

**Division of responsability between the library and extension code proper:**
The extension has to provide the library with one main thing: an executable or a script that can listen for messages sent on a given port and translate them into `eval`-able statements. For the python extension, this script is `pyext.py`. That script listens on a localhost socket for messages that tell it what code to `eval` and then sends it back. Most everything else that the extension code itself has to do is boilerplate. Read the documentation of the library code for more.

### CODE REVIEW

**Comment by qiemem** [comment]
Hey just want say, as the original author of this extension, this looks super rad and I'm really excited this work happening. Feel free to reach out to me if you have any questions or anything, though glancing through the code, it looks like you've already figured it all out!

File: src/main/ConfigDialog.scala [hunk 1]
```diff
@@ -94,4 +94,7 @@ class PythonMenu extends JMenu("Python") {
   add("Configure").addActionListener{ _ =>
     new ConfigEditor(App.app.frame, PythonExtension.config).setVisible(true)
   }
+  add("Pop-out Interpreter").addActionListener{ _ =>
```
Comment (LaCuneta) [inline src/main/ConfigDialog.scala:97]: I don't love the name `Pop-out Interpreter`.  Most NetLogo-ish would be `Open Python Command Center`, but you could talk me into something like `Interactive Python Console`.  What do you think?

**Review by LaCuneta — CHANGES_REQUESTED** [review]
Hey, this is great!  As we probably should have guessed, given the nature of the changes, a lot of my feedback is around the language library itself.  I'll open issues in its repo instead of cluttering up this PR.

I always love to see PRs removing more code than they add!  The code all looks good, I just a few small style niggles left as comments.

Should we add a note to the documentation about the interactive command window?

The Python extension is one a few setup to run tests standalone without being a submodule of the NetLogo repo; you can just run `sbt test` in this repo and it'll copy files around to get the NetLogo language test running to go.  Those tests aren't passing on this branch, but I'm not sure that has anything to do with this PR directly.  I'll look into this, and update the extension to run tests through Actions so we get that feedback faster in the future.

File: src/main/ConfigDialog.scala (hunk 1 above)
Comment (JamesHovet) [reply src/main/ConfigDialog.scala:97]: My gut is that the Python extension is for people who want to make netlogo models but get frustrated enough with doing things in vanilla netlogo that they turn to a more familiar language like Python to do some heavy lifting. To me these people would be more familiar with the non-netlogo terminology than they would be the command center. I forgot what the command center was for a minute when I first read this honestly.

**Review by LaCuneta — APPROVED** [review]

**Pull request merged**